lambeq/text2diagram/pregroup_tree_converter.py

```diff
@@ -740,6 +740,43 @@ def tree2diagram(
             if other_node_ind == -1:
                 free_wires = list(subword_ty_global_inds)
 
+    # Checks for the type `n.r @ s @ ...` for a word and
```

Comment:
@neiljdo Let's put this code into a (private) method or something and call it here? The handling of this case shouldn't interact so much with the conversion logic, we might add more special cases later.